library/core/src/ops/unsize.rs

```diff
@@ -32,41 +32,78 @@ use crate::marker::Unsize;
 /// [unsize]: crate::marker::Unsize
 /// [nomicon-coerce]: ../../nomicon/coercions.html
 #[unstable(feature = "coerce_unsized", issue = "27732")]
+#[cfg_attr(not(bootstrap), lang = "unsafe_coerce_unsized")]
+pub trait UnsafeCoerceUnsized<T: ?Sized> {
+    // Empty.
+}
+
+/// Trait that indicates that this is a pointer or a wrapper for one,
+/// where unsizing can be performed on the pointee.
+///
+/// See the [DST coercion RFC][dst-coerce] and [the nomicon entry on coercion][nomicon-coerce]
+/// for more details.
+///
+/// When this trait is implemented in addition to [`UnsafeCoerceUnsized`](unsafe-coerce-unsized),
```

Comment:
While I understand the reasoning behind it, I think it's confusing to have`CoerceUnsized` be an unsafe trait, and `UnsafeCoerceUnsized` be a safe trait.

We've used 'unchecked' in the names of unsafe functions (e.g. `get_unchecked`) - I think it might be clearer to rename `UnsafeCoerceUnsized` to `UncheckedCoerceUnsized`.

Reply:
Sounds good, thanks!